Build failure: mpv

### Steps To Reproduce

Steps to reproduce the behavior:

1. Fetch latest `master`
1. Run `nix-build -A pkgs.mpv`

### Build log

```
FAILED: osdep/mac/swift.swiftmodule osdep/mac/swift.h osdep/mac/swift.o 
/nix/store/mms5vi0wxy511i9faihkrzalzpfk0gfi-swift-wrapper-5.8/bin/swift -frontend -c -sdk /nix/store/chm77nb9a51ys9hkwybv9gydai8cjz7c-SDKs/MacOSX11.0.sdk -enable-objc-interop -emit-objc-header -parse-as-library -O -D HAVE_MACOS_10_15_4_FEATURES -D HAVE_MACOS_11_FEATURES -D HAVE_MACOS_COCOA_CB -D HAVE_MACOS_TOUCHBAR -D HAVE_MACOS_MEDIA_PLAYER -module-name swift -emit-module-path osdep/mac/swift.swiftmodule -import-objc-header /private/tmp/nix-build-mpv-0.39.0.drv-0/source/osdep/mac/app_bridge_objc.h -emit-objc-header-path osdep/mac/swift.h -o osdep/mac/swift.o ../osdep/mac/application.swift ../osdep/mac/app_hub.swift ../osdep/mac/event_helper.swift ../osdep/mac/input_helper.swift ../osdep/mac/libmpv_helper.swift ../osdep/mac/log_helper.swift ../osdep/mac/menu_bar.swift ../osdep/mac/option_helper.swift ../osdep/mac/precise_timer.swift ../osdep/mac/presentation.swift ../osdep/mac/swift_compat.swift ../osdep/mac/swift_extensions.swift ../osdep/mac/type_helper.swift ../video/out/mac/common.swift ../video/out/mac/title_bar.swift ../video/out/mac/view.swift ../video/out/mac/window.swift ../video/out/cocoa_cb_common.swift ../video/out/mac/gl_layer.swift ../osdep/mac/remote_command_center.swift ../osdep/mac/touch_bar.swift -I. -I/private/tmp/nix-build-mpv-0.39.0.drv-0/source -I/nix/store/409zncmgji52kg643w8b2fymn0zl9hmp-libplacebo-7.349.0/include -I/nix/store/10ylmxvkisc5l75n0m0lmhr2hw3cyn68-ffmpeg-6.1.2-dev/include
../osdep/mac/input_helper.swift:77:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
```

### Additional context

It was reported originally in https://github.com/mpv-player/mpv/issues/13763, it seems to be an issue with the version of Swift that's being used.


### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@AndersonTorres
@fpletz
@globin
@ma27


### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.0.0, macOS 15.0`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixpkgs"`
 - channels(user): `"home-manager, nixpkgs"`
 - nixpkgs: `~/.nix-defexpr/channels/nixpkgs`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/video/mpv/0001-fix-darwin-build.patch
+++ b/pkgs/applications/video/mpv/0001-fix-darwin-build.patch
@@ -1,0 +1,19 @@
+diff --git a/osdep/mac/input_helper.swift b/osdep/mac/input_helper.swift
+index 0acec6bd40..0ec5837864 100644
+--- a/osdep/mac/input_helper.swift
++++ b/osdep/mac/input_helper.swift
+@@ -18,6 +18,14 @@
+ import Cocoa
+ import Carbon.HIToolbox
+ 
++extension NSCondition {
++    fileprivate func withLock<T>(_ body: () throws -> T) rethrows -> T {
++        self.lock()
++        defer { self.unlock() }
++        return try body()
++    }
++}
++
+ class InputHelper: NSObject {
+     var option: OptionHelper?
+     var lock = NSCondition()

--- a/pkgs/applications/video/mpv/default.nix
+++ b/pkgs/applications/video/mpv/default.nix
@@ -151,6 +151,11 @@ stdenv'.mkDerivation (finalAttrs: {
     hash = "sha256-BOGh+QBTO7hrHohh+RqjSF8eHQH8jVBPjG/k4eyFaaM=";
   };
 
+  patches = [
+    # Fix build with Darwin SDK 11
+    ./0001-fix-darwin-build.patch
+  ];
+
   postPatch = lib.concatStringsSep "\n" [
     # Don't reference compile time dependencies or create a build outputs cycle
     # between out and dev
@@ -308,6 +313,8 @@ stdenv'.mkDerivation (finalAttrs: {
       pushd ../TOOLS
       cp mpv_identify.sh umpv $out/bin/
       popd
+    ''
+    + lib.optionalString stdenv.hostPlatform.isLinux ''
       pushd $out/share/applications
 
       sed -e '/Icon=/ ! s|mpv|umpv|g; s|^Exec=.*|Exec=umpv %U|' \

--- a/pkgs/servers/gonic/default.nix
+++ b/pkgs/servers/gonic/default.nix
@@ -50,6 +50,9 @@ buildGoModule rec {
     tests.gonic = nixosTests.gonic;
   };
 
+  # tests require it
+  __darwinAllowLocalNetworking = true;
+
   meta = {
     homepage = "https://github.com/sentriz/gonic";
     description = "Music streaming server / subsonic server API implementation";

--- a/pkgs/servers/gonic/default.nix
+++ b/pkgs/servers/gonic/default.nix
@@ -30,18 +30,18 @@ buildGoModule rec {
   postPatch = lib.optionalString transcodingSupport ''
     substituteInPlace \
       transcode/transcode.go \
-      --replace \
+      --replace-fail \
         '`ffmpeg' \
         '`${lib.getBin ffmpeg}/bin/ffmpeg'
   '' + ''
     substituteInPlace \
       jukebox/jukebox.go \
-      --replace \
+      --replace-fail \
         '"mpv"' \
         '"${lib.getBin mpv}/bin/mpv"'
   '' + ''
     substituteInPlace server/ctrlsubsonic/testdata/test* \
-      --replace \
+      --replace-quiet \
         '"audio/flac"' \
         '"audio/x-flac"'
   '';

